client/client-multi/client-objective-c/KaaTests/logging/storage/CommonLogStorageTests.m

```diff
@@ -0,0 +1,341 @@
+        NSInteger leftSize = [[leftBucket1 logRecords] count];
+        
+        if (leftBucket2)
+            leftSize += [[leftBucket2 logRecords] count];
+        
+        XCTAssertEqual(leftSize, insertionCount, @"Failue for storage: %@", [(id)storage class]);
+        
+        [storage close];
+    }
+}
+
+- (void)testLogStorageCountAndVolume {
+    int64_t bucketSize = 9;
+    int32_t recordCount = 3;
+    
+    NSArray *storagesToTest = [self storagesToTestWithBucketSize:bucketSize recordCount:recordCount];
+    
+    for (id<LogStorage> storage in storagesToTest) {
+        LogRecord *record = [LogTestHelper defaultLogRecord];
+        
+        int32_t insertionCount = 9;
+        int32_t receivedCount = 0;
+        int32_t iter = insertionCount;
+        
+        while (iter-- > 0) {
+            [storage addLogRecord:record];
+        }
+        
+        LogBucket *logBucket = [storage getNextBucket];
+        receivedCount = [self addRecordCountIfNotEmpty:receivedCount toLogBucket:logBucket];
+        XCTAssertEqual(insertionCount - receivedCount, [[storage getStatus] getRecordCount],
+                       @"Failue for storage: %@", [(id)storage class]);
+        XCTAssertEqual((insertionCount - receivedCount) * RECORD_PAYLOAD_SIZE, [[storage getStatus] getConsumedVolume],
+                       @"Failue for storage: %@", [(id)storage class]);
+        
+        logBucket = [storage getNextBucket];
+        receivedCount = [self addRecordCountIfNotEmpty:receivedCount toLogBucket:logBucket];
+        XCTAssertEqual(insertionCount - receivedCount, [[storage getStatus] getRecordCount],
+                       @"Failue for storage: %@", [(id)storage class]);
+        XCTAssertEqual((insertionCount - receivedCount) * RECORD_PAYLOAD_SIZE, [[storage getStatus] getConsumedVolume],
+                       @"Failue for storage: %@", [(id)storage class]);
+        
+        logBucket = [storage getNextBucket];
+        receivedCount = [self addRecordCountIfNotEmpty:receivedCount toLogBucket:logBucket];
+        XCTAssertEqual(insertionCount - receivedCount, [[storage getStatus] getRecordCount],
+                       @"Failue for storage: %@", [(id)storage class]);
+        XCTAssertEqual((insertionCount - receivedCount) * RECORD_PAYLOAD_SIZE, [[storage getStatus] getConsumedVolume],
+                       @"Failue for storage: %@", [(id)storage class]);
+        
+        [storage rollbackBucketWithId:[logBucket bucketId]];
+        receivedCount -= [[logBucket logRecords] count];
+        XCTAssertEqual(insertionCount - receivedCount, [[storage getStatus] getRecordCount],
+                       @"Failue for storage: %@", [(id)storage class]);
+        XCTAssertEqual((insertionCount - receivedCount) * RECORD_PAYLOAD_SIZE, [[storage getStatus] getConsumedVolume],
+                       @"Failue for storage: %@", [(id)storage class]);
+    }
+}
+
+- (void)helpAddingRecords:(int32_t)addedCount
```

Comment:
"Records" in method name seems to be actual records(like array). Need to add word "Count"